pkg/cloudops/gcp/gcp.go

```diff
@@ -0,0 +1,227 @@
+}
+
+// NewGCPOps returns an instance of the GCP cloud ops implementation
+func NewGCPOps() (cloudops.Ops, error) {
+	ops := &gcpOps{
+		instInfo: &cloudops.InstanceInfo{},
+	}
+	err := ops.populateBasicInfo()
+	if err != nil {
+		return nil, err
+	}
+
+	ops.instInfo.Region = ops.instInfo.Zone[:len(ops.instInfo.Zone)-2]
+
+	ctx := context.Background()
+	ops.computeService, err = compute.NewService(ctx, option.WithScopes(compute.ComputeScope))
+	if err != nil {
+		return nil, fmt.Errorf("unable to create Compute service: %v", err)
+	}
+
+	ops.containerService, err = container.NewService(ctx, option.WithScopes(compute.CloudPlatformScope))
+	if err != nil {
+		return nil, fmt.Errorf("unable to create Container service: %v", err)
+	}
+
+	return ops, nil
+}
+
+func (g *gcpOps) InspectSelf() (*cloudops.InstanceInfo, error) {
+	inst, err := g.computeService.Instances.Get(g.project, g.instInfo.Zone, g.instInfo.Name).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	g.instInfo.Labels = inst.Labels
+	return g.instInfo, nil
+}
+
+func (g *gcpOps) InspectSelfInstanceGroup() (*cloudops.InstanceGroupInfo, error) {
+	inst, err := g.computeService.Instances.Get(g.project, g.instInfo.Zone, g.instInfo.Name).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	meta := inst.Metadata
+	if meta == nil {
+		return nil, fmt.Errorf("instance doesn't have metadata set")
+	}
+
+	var (
+		gkeClusterName   string
+		instanceTemplate string
+		clusterLocation  string
+	)
+
+	for _, item := range meta.Items {
+		if item == nil {
+			continue
+		}
+
+		if item.Key == clusterNameKey {
+			if item.Value == nil {
+				return nil, fmt.Errorf("instance has %s key in metadata but has invalid value", clusterNameKey)
+			}
+
+			gkeClusterName = *item.Value
+		}
+
+		if item.Key == instanceTemplateKey {
+			if item.Value == nil {
+				return nil, fmt.Errorf("instance has %s key in metadata but has invalid value", instanceTemplateKey)
+			}
+
+			instanceTemplate = *item.Value
+		}
+
+		if item.Key == clusterLocationKey {
+			if item.Value == nil {
+				return nil, fmt.Errorf("instance has %s key in metadata but has invalid value", clusterLocationKey)
+			}
+
+			clusterLocation = *item.Value
+		}
+	}
+
+	if len(gkeClusterName) == 0 || len(instanceTemplate) == 0 || len(clusterLocation) == 0 {
+		return nil, fmt.Errorf("API is currently only supported on the GKE platform.")
+	}
+
+	for _, item := range meta.Items {
+		if item == nil {
+			continue
+		}
+
+		if item.Key == kubeLabelsKey {
+			if item.Value == nil {
+				return nil, fmt.Errorf("instance has %s key in metadata but has invalid value", kubeLabelsKey)
+			}
+
+			kubeLabels, err := parser.LabelsFromString(*item.Value)
+			if err != nil {
+				return nil, err
+			}
+
+			for labelKey, labelValue := range kubeLabels {
+				if labelKey == nodePoolKey {
+					nodePoolPath := fmt.Sprintf("projects/%s/locations/%s/clusters/%s/nodePools/%s", "portworx-eng", clusterLocation, gkeClusterName, labelValue)
```

Comment:
nit: split the fmt.Sprintf into multiple lines. It is just easier to read

Reply:
done